src/filter/http_context.rs

```diff
@@ -114,15 +127,15 @@ impl Context for Filter {
 
```

Comment:
I think we should check `status_code`. When grpc endpoint is unreachable, this is flagged on the `status_code`.

Reply:
it might be useful, however I'm a bit confused about what are the `status_code` valid values... since there's no docs and it's confusing their examples... would it be a [GrpcStatusCode](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/blob/main/src/types.rs#L121) or [Status](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/blob/main/src/types.rs#L43).. then it's a bit confusing in this [particular example](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/blob/main/examples/grpc_auth_random/src/lib.rs#L70-L80) how they treat it. What do you reckon? what meaningful check can we do to avoid parsing and returning early?

Reply:
From the tests I have run, on successful GRPC request/response exchange, `status_code` is [0](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/blob/main/src/types.rs#L122). Whereas when the service is unreachable, `status_code` is [14](https://github.com/proxy-wasm/proxy-wasm-rust-sdk/blob/main/src/types.rs#L136).

That all I know for now.

Let's merge this and you can enhance this in a following up PR's